Microsoft.Toolkit.Uwp.UI.Controls/ImageCropper/ImageCropper.Properties.cs

```diff
@@ -0,0 +1,171 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Media.Imaging;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// The <see cref="ImageCropper"/> control allows user to crop image freely.
+    /// </summary>
+    public partial class ImageCropper
+    {
+        /// <summary>
+        /// Gets or sets the minimum cropped length(in pixel).
+        /// </summary>
+        public double MinCroppedPixelLength { get; set; } = 40;
+
+        /// <summary>
+        /// Gets or sets the minimum selectable length.
+        /// </summary>
+        public double MinSelectedLength { get; set; } = 40;
+
+        private static void OnSourceImageChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var target = (ImageCropper)d;
+            if (e.NewValue is WriteableBitmap bitmap)
+            {
+                if (bitmap.PixelWidth > target.MinCropSize.Width && bitmap.PixelHeight > target.MinCropSize.Height)
+                {
+                    target.InitImageLayout();
+                }
+                else
+                {
+                    throw new ArgumentException("The resolution of the image is too small!");
+                }
+            }
+        }
+
+        private static void OnAspectRatioChanged(
+            DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var target = (ImageCropper)d;
+            target.UpdateAspectRatio();
+        }
+
+        private static void OnCircularCropChanged(
+            DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var target = (ImageCropper)d;
+            target.UpdateAspectRatio();
+            target.UpdateControlButtonVisibility();
+            target.UpdateMaskArea();
+        }
+
+        private static void OnIsSecondaryControlButtonVisibleChanged(
+            DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var target = (ImageCropper)d;
+            target.UpdateControlButtonVisibility();
+        }
+
+        /// <summary>
+        ///  Gets or sets the source of the cropped image.
+        /// </summary>
+        public WriteableBitmap SourceImage
```

Comment:
Also does it have to be a WritableBitmap vs. an ImageSource type?

Reply:
It should be the `Source`,but we can't use `ImageSource` instead of `WritableBitmap` .
If an `ImageSource` is used, there may be no way to get the image metadata.